service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/storageFolder/CreateAwsStorageFolderStep.java

```diff
@@ -0,0 +1,87 @@
+package bio.terra.workspace.service.resource.controlled.cloud.aws.storageFolder;
+
+import bio.terra.common.exception.ConflictException;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.AwsUtils;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.workspace.AwsCloudContextService;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sts.model.Tag;
+
+public class CreateAwsStorageFolderStep implements Step {
+  private static final Logger logger = LoggerFactory.getLogger(CreateAwsStorageFolderStep.class);
+  private final ControlledAwsStorageFolderResource resource;
+  private final AwsCloudContextService awsCloudContextService;
+  private final AuthenticatedUserRequest userRequest;
+
+  public CreateAwsStorageFolderStep(
+      ControlledAwsStorageFolderResource resource,
+      AwsCloudContextService awsCloudContextService,
+      AuthenticatedUserRequest userRequest) {
+    this.resource = resource;
+    this.awsCloudContextService = awsCloudContextService;
+    this.userRequest = userRequest;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    AwsCredentialsProvider credentialsProvider =
+        AwsUtils.createWsmCredentialProvider(
+            awsCloudContextService.getRequiredAuthentication(),
+            awsCloudContextService.discoverEnvironment());
+
+    if (AwsUtils.checkFolderExists(
```

Comment:
Doesn't this need to be in a separate step if it's going to be fatal?  If the subsequent folder creation succeeds, but the flight does not finish, this will fail the flight on retry.

Reply:
Makes sense, changed